<launch>

    <arg name="external_manager" default="false"/>
    <arg name="manager_name" default="celex5_mipi_manager"/>
    <arg name="required" default="false"/>

    <arg name="tf_prefix" default=""/>

    <!--           CeleX5-MIPI Mode             -->
    <!--  Event_Off_Pixel_Timestamp_Mode    0   -->
    <!--  Event_In_Pixel_Timestamp_Mode     1   -->
    <!--  Event_Intensity_Mode              2   -->
    <!--  Full_Picture_Mode                 3   -->
    <!--  Optical_Flow_Mode                 4   -->
    <!--  Optical_Flow_FPN_Mode             5   -->
    <!--  Multi_Read_Optical_Flow_Mode      6   -->

    <arg name="fixed_mode" default="0"/>
    <arg name="loop_mode1" default="1"/>
    <arg name="loop_mode2" default="3"/>
    <arg name="loop_mode3" default="4"/>

    <arg name="event_frame_time" default="1000"/>
    <arg name="optical_flow_frame_time" default="20"/>
    <arg name="threshold" default="185"/>
    <arg name="brightness" default="150"/>
    <!--    <arg name="contrast" default="1"/>-->
    <arg name="clock_rate" default="100"/>

    <arg name="is_loop_mode_enabled" default="false"/>
    <arg name="event_duration_in_loop" default="100"/>
    <arg name="picture_number_in_loop" default="30"/>

    <arg name="event_FPN_file_path" default="config/FPN_3.txt"/>
    <arg name="frame_FPN_file_path" default="config/FPN_Optical.txt"/>

    <node unless="$(arg external_manager)" pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager"
          output="screen" required="$(arg required)"/>
    <node pkg="nodelet" type="nodelet" name="celex5_mipi" args="load celex5_ros/CeleX5Nodelet $(arg manager_name)"
          required="$(arg required)">
        <param name="fixed_mode" type="int" value="$(arg fixed_mode)"/>
        <param name="loop_mode1" type="int" value="$(arg loop_mode1)"/>
        <param name="loop_mode2" type="int" value="$(arg loop_mode2)"/>
        <param name="loop_mode3" type="int" value="$(arg loop_mode3)"/>
        <param name="event_frame_time" type="int" value="$(arg event_frame_time)"/>
        <param name="optical_flow_frame_time" type="int" value="$(arg optical_flow_frame_time)"/>
        <param name="threshold" type="int" value="$(arg threshold)"/>
        <param name="brightness" type="int" value="$(arg brightness)"/>
        <param name="clock_rate" type="int" value="$(arg clock_rate)"/>
        <param name="is_loop_mode_enabled" type="bool" value="$(arg is_loop_mode_enabled)"/>
        <param name="event_duration_in_loop" type="int" value="$(arg event_duration_in_loop)"/>
        <param name="picture_number_in_loop" type="int" value="$(arg picture_number_in_loop)"/>
        <param name="event_FPN_file_path" type="str" value="$(arg event_FPN_file_path)"/>
        <param name="frame_FPN_file_path" type="str" value="$(arg frame_FPN_file_path)"/>
    </node>

</launch>